### ISSUE TITLE
Lost messages in Closed Group

### ISSUE THREAD

**Issue by Sepero** [oxen-io/session-android#713]
This is a bad bug because it's difficult to isolate (Issue with Android? Or issue with server nodes? Or other? Are messages also being lost in 1 to 1 chats?)

I wouldn't have known this bug existed, except for the fact I have a closed group with my spouse and some family.

What happened:

1. A normal text message (not a reply, not a url, not an image) was sent by one of my family to our Closed Group.
2. The sender was running Android Session v1.11.6
3. My spouse and I both running Android Session v1.11.8
4. I could see the message received on my spouse's device, but not mine.
5. I waited over 24 hours, but the message never came to my device.

How to replicate, unknown

**Comment by hjubb** [comment]
Will be looking into this and trying to replicate it, as you mentioned it's hard to track down exactly what or where it's happening but will keep checking if there's a way to eliminate this. If you do find a way to reliably reproduce it that would be super handy but will continue to try to fix this otherwise and keep posted on this issue :+1:

**Comment by KeeJef** [comment]
out of curiosity you using Slow or Fast mode for notifications?

**Comment by Sepero** [comment]
> out of curiosity you using Slow or Fast mode for notifications?

Interesting question that. I switch it from time to time depending on the need. Usually Slow mode

**Comment by Sepero** [comment]
An example conversation from 2 separate devices. The missing part highlighted in soft red

First Device
![Screenshot_20211010-155311__01](https://user-images.githubusercontent.com/1575546/136702540-5a910f8a-7b19-4437-a38c-25984de37caa.jpg)

----
Second Device
![session-2021-10-10-221157__01](https://user-images.githubusercontent.com/1575546/136702545-605d3949-14fa-48c5-b15b-27af1eeeb14a.jpg)

**Comment by hjubb** [comment]
Thanks for additional info @Sepero

**Comment by Sepero** [comment]
Just as an additional remark, the second post with images is from a different closed group. I don't know if this bug is specific to receiving on Android, or if it affects all platforms

**Comment by Sepero** [comment]
![Screenshot_20211119-151934](https://user-images.githubusercontent.com/1575546/142603609-b865b188-8b07-4763-bf2a-66c23b314511.jpg)

![Screenshot_20211119-151915](https://user-images.githubusercontent.com/1575546/142603671-6f4d04b2-f45b-4cb5-a36a-f725573afbd5.jpg)


Still occurring v1.11.12. Above are two more examples from recent chats.

**Comment by Sepero** [comment]
![Screenshot_20211208-210933](https://user-images.githubusercontent.com/1575546/145223828-cb9b0341-fa83-4f9e-ad90-886f3cc11217.jpg)

An entire conversation lost. Session v1.11.13

**Comment by Sepero** [comment]
v1.11.14 still losing messages

About a week ago, I did took some steps to help me see the issue better.
1. Uninstalled and reinstalled with my recovery phrase to see if that would correct the issue
2. Installed the desktop client with a new account, and invited that account into the closed groups.

Unfortunately messages are still being lost. I am experiencing this in multiple closed groups. Messages are showing up on the desktop client that do not appear on my mobile.

Image Android
![Screenshot_20211226-113402__01](https://user-images.githubusercontent.com/1575546/147399023-8da2c643-de4b-42b7-bbfc-d1d4ba202ebc.jpg)


Image Desktop
![Screenshot_20211226-112924__01__01](https://user-images.githubusercontent.com/1575546/147399026-036a05f2-0da9-49e5-8993-d14b47e7bb6d.jpg)

**Comment by Sepero** [comment]
2 people leaving a Closed Group shown on Desktop client. No message of them leaving on Android.

2 people leaving the group on Desktop client.
![Screenshot_20220116-180754](https://user-images.githubusercontent.com/1575546/149669742-fb84a008-54be-4b94-bc31-ff012f4dc113.jpg)

No message of them leaving on Android.
![Screenshot_20220116-180828](https://user-images.githubusercontent.com/1575546/149669751-0f9a789e-4531-47db-bc8c-3401c35069aa.jpg)

**Comment by Sepero** [comment]
When I opened this issue, I started with this

> This is a bad bug because it's difficult to isolate (Issue with Android? Or issue with server nodes? Or other? Are messages also being lost in 1 to 1 chats?)

Now I believe it is an Android only issue, and only affects Closed Groups. I've seen no signs of losing Open Group nor 1 to 1 messages on Android

**Comment by hjubb** [comment]
That's excellent info thanks @Sepero it might have something to do with the closed group specific poller or maybe exchanging new encryption keys after group members are removed

**Comment by lennylen** [comment]
This might be protocol issue as it happens on linux desktop, android, iOS as well for me. The application doesn't "catch" the occasional message, creating a broken experience.

**Comment by Sepero** [comment]
> This might be protocol issue as it happens on linux desktop, android, iOS as well for me. The application doesn't "catch" the occasional message, creating a broken experience.

It's possible there could be a protocol issue, And an Android issue. Though personally, I have not had indication of even 1 lost message on Desktop client. I'm using separate accounts (not syncing)

@lennylen Are you syncing a single account? If so, then it could be a sync issue.

**Comment by lennylen** [comment]
> > This might be protocol issue as it happens on linux desktop, android, iOS as well for me. The application doesn't "catch" the occasional message, creating a broken experience.
> 
> 
> 
> It's possible there could be a protocol issue, And an Android issue. Though personally, I have not had indication of even 1 lost message on Desktop client. I'm using separate accounts (not syncing)
> 
> 
> 
> @lennylen Are you syncing a single account? If so, then it could possibly be a sync issue.

Nope, all separate accounts. I have had missed messages on desktop the past few versions.

**Comment by Sepero** [comment]
Very interesting.

When I say I've not received any indication of missing messages on Desktop, it's because every message I've receive on Android, I also received on Desktop. Though, I have messages not being received on Android, but are being received on Desktop (inconsistent delivery).

@lennylen Are you are experiencing missing messages consistently across all platforms, or inconsistently like the issue here?

**Comment by Sepero** [comment]
Unresolved v1.11.16

**Comment by KeeJef** [comment]
When were the closed groups created where this issue is present? Is this occurring in newly created closed groups or just ones that were created with very old versions of Session? Wondering if there may have been an issue when we migrated to new closed group types later last year.

**Comment by Sepero** [comment]
> When were the closed groups created where this issue is present? Is this occurring in newly created closed groups or just ones that were created with very old versions of Session? Wondering if there may have been an issue when we migrated to new closed group types later last year.

This issue was first identified around Session v1.11.6, and this issue is occurring in groups created after that version.

**Comment by Sepero** [comment]
1. Screenshot of member leaving Closed Group on Linux Desktop client.
2. Android client did not show the member leave, and member remains in group list. Android client is group creator.
3. Desktop client and Android client are separate accounts.

![Screenshot_20220304-115623](https://user-images.githubusercontent.com/1575546/156702640-b284e343-a3f0-40a7-a77b-d2f2482fe607.jpg)

**Comment by Sepero** [comment]
A different closed group, not owned by myself. Another member leaves. Shown on Desktop Linux. Nothing shown on Android

![Screenshot_20220306-023748](https://user-images.githubusercontent.com/1575546/156897755-5a58e1a6-bc2a-456e-b26d-003dd766f7c5.jpg)

*Android still shows member in group list.
![Screenshot_20220306-024053__01](https://user-images.githubusercontent.com/1575546/156897897-0eba4fae-586e-4721-8db0-5b33727f54ca.jpg)

**Comment by hjubb** [comment]
This particular case looks like it could be related to something intermittent between keys rotating on people leaving, it definitely gives a better idea of where it's messing up, thanks for this extra info

**Comment by Sepero** [comment]
Additional info.
I'm not 100% certain, but I believe I've also experienced undelivered messages in closed groups where no members have ever exited.

**Comment by Sepero** [comment]
> When I say I've not received any indication of missing messages on Desktop, it's because every message I've receive on Android, I also received on Desktop. Though, I have messages not being received on Android, but are being received on Desktop (inconsistent delivery).

UPDATE:
I have recently noticed missing messages on Desktop. Comparing messages on Desktop and Mobile in same groups, I found some messages with both

- I have messages that were received on Android that were not received on Desktop
- I have messages that were received on Desktop that were not received on Android

**Comment by Sepero** [comment]
Has this big been reported on Desktop yet? I was unable to find it. Possibly it is a bug in the protocol as @lennylen suggested.
